hypothesis-python/src/hypothesis/internal/conjecture/shrinking/floats.py

```diff
@@ -82,6 +76,10 @@ def run_step(self):
 
         for p in range(10):
             scaled = self.current * 2**p  # note: self.current may change in loop
+            # floats close to math.inf can overflow in this intermediate step.
+            # probably something we should fix?
```

Comment:
This part (like the fraction shrink below) only matter when the fractional part is nonzero, i.e. for floats less than 2^53. So the test for `inf` is in practice fine, but it may be clearer/better to make the loop conditional, or move the integer delegation inside the loop so that it triggers before it has a chance to overflow.

Reply:
thanks. I think fdeeb0d40986c4f3abace91e419de08198474e72 makes this clearer